frame/dapps-staking/src/traits.rs

```diff
@@ -0,0 +1,14 @@
+use super::*;
+
+/// Multi-VM pointer to smart contract instance.
+#[derive(PartialEq, Eq, Copy, Clone, Encode, Decode, RuntimeDebug)]
+pub enum SmartContract<AccountId> {
+    /// EVM smart contract instance.
+    Evm(sp_core::H160),
+    /// Wasm smart contract instance.
+    Wasm(AccountId),
+}
```

Comment:
You should move this to the runtime implementation.

dapps-staking pallet doesn't need to know about such details.